controllers/replication.storage/volumegroupreplication_controller.go

```diff
@@ -33,21 +56,533 @@ type VolumeGroupReplicationReconciler struct {
+				err := r.setGroupReplicationFailure(instance, logger, err)
+				return reconcile.Result{}, err
+			}
+
+			// Create/Fetch VolumeGroupReplicationContent CR
+			if instance.Spec.VolumeGroupReplicationContentName != "" {
+				err = r.Client.Get(ctx, types.NamespacedName{Name: instance.Spec.VolumeGroupReplicationContentName}, vgrContentObj)
+				if err != nil {
+					logger.Error(err, "failed to get volumeGroupReplicationContent resource", "VGRContentName", instance.Spec.VolumeGroupReplicationContentName)
+					err := r.setGroupReplicationFailure(instance, logger, err)
+					return reconcile.Result{}, err
+				}
+				if !reflect.DeepEqual(pvHandlesList, vgrContentObj.Spec.Source.VolumeHandles) {
+					vgrContentObj.Spec.Source.VolumeHandles = pvHandlesList
+					err = r.Client.Update(ctx, vgrContentObj)
+					if err != nil {
+						logger.Error(err, "failed to update volumeGroupReplicationContent resource", "VGRContentName", vgrContentObj.Name)
+						err := r.setGroupReplicationFailure(instance, logger, err)
+						return reconcile.Result{}, err
+					}
+				}
+			} else {
+				vgrContentObj = &replicationv1alpha1.VolumeGroupReplicationContent{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("vgrcontent-%s", instance.UID),
+						Annotations: map[string]string{
+							"replication.storage.openshift.io/volumegroupref": fmt.Sprintf("%s/%s", instance.Name, instance.Namespace),
+						},
+					},
+					Spec: replicationv1alpha1.VolumeGroupReplicationContentSpec{
+						VolumeGroupReplicationRef: corev1.ObjectReference{
+							APIVersion: instance.APIVersion,
+							Kind:       instance.Kind,
+							Name:       instance.Name,
+							Namespace:  instance.Namespace,
+							UID:        instance.UID,
+						},
+						Provisioner:                     vgrClassObj.Spec.Provisioner,
+						VolumeGroupReplicationClassName: instance.Spec.VolumeGroupReplicationClassName,
+						Source: replicationv1alpha1.VolumeGroupReplicationContentSource{
+							VolumeHandles: pvHandlesList,
+						},
+					},
+				}
+
+				err = r.Client.Create(ctx, vgrContentObj)
+				if err != nil {
+					logger.Error(err, "failed to create volumeGroupReplicationContent", "VGRContentName", vgrContentObj.Name)
+					err := r.setGroupReplicationFailure(instance, logger, err)
+					return reconcile.Result{}, err
+				}
+
+				// Update the VGR with VGRContentName
+				instance.Spec.VolumeGroupReplicationContentName = vgrContentObj.Name
+				err = r.Client.Update(ctx, instance)
+				if err != nil {
+					logger.Error(err, "failed to update volumeGroupReplication instance", "VGRName", instance.Name)
+					err := r.setGroupReplicationFailure(instance, logger, err)
+					return reconcile.Result{}, err
+				}
+			}
+		}
+
+		if vgrContentObj.Spec.VolumeGroupReplicationHandle == "" {
+			logger.Info("Either volumegroupreplicationcontent is not yet created or it is still grouping the volumes to be replicated")
+			return reconcile.Result{}, nil
```

Comment:
requeue this one or else if wont get reconciled again?

Reply:
@Madhu-1 I was wondering if we should be issuing reconcile here, since anyway we are watching the VGRContent resource, so when the grouping is done the VGRContent would be updated with the groupreplicationhandle, and that would anyway trigger a reconcile fo the VGR resource. So, what's the benefit of requeuing here?